Enable LDFLAGS when building ovnk binaries

zshi-redhat opened the pull request:
LDFLAGS="-w -s" reduces ~10M size for each ovnk binary

w/ LDFLAGS="-w -s":
-rwxrwxr-x. 1 root root  37M Feb  7 10:20 hybrid-overlay-node 
-rwxrwxr-x. 1 root root  35M Feb  7 10:20 ovndbchecker 
-rwxrwxr-x. 1 root root  35M Feb  7 10:20 ovn-k8s-cni-overlay 
-rwxrwxr-x. 1 root root  45M Feb  7 10:20 ovnkube
-rwxrwxr-x. 1 root root  30M Feb  7 10:21 ovnkube-trace 
-rwxrwxr-x. 1 root root  31M Feb  7 10:20 ovn-kube-util

w/o LDFLAGS="-w -s":

-rwxrwxr-x. 1 root root  52M Feb  7 10:28 hybrid-overlay-node 
-rwxrwxr-x. 1 root root  49M Feb  7 10:28 ovndbchecker 
-rwxrwxr-x. 1 root root  50M Feb  7 10:28 ovn-k8s-cni-overlay 
-rwxrwxr-x. 1 root root  64M Feb  7 10:28 ovnkube
-rwxrwxr-x. 1 root root  42M Feb  7 10:28 ovnkube-trace 
-rwxrwxr-x. 1 root root  44M Feb  7 10:28 ovn-kube-util

Signed-off-by: Zenghui Shi <zshi@redhat.com>

trozet commented:
I'm not sure we can do this, from what I found:

```
You will get the smallest binaries if you compile with -ldflags '-w -s'.

The -w turns off DWARF debugging information: you will not be able to use gdb on the binary to look at specific functions or set breakpoints or get stack traces, because all the metadata gdb needs will not be included. You will also not be able to use other tools that depend on the information, like pprof profiling.

The -s turns off generation of the Go symbol table: you will not be able to use go tool nm to list the symbols in the binary. strip -s is like passing -s to -ldflags but it doesn't strip quite as much. go tool nm might still work after strip -s. I am not completely sure.

None of these — not -ldflags -w, not -ldflags -s, not strip -s — should affect the execution of the actual program. They only affect whether you can debug or analyze the program with other tools.
```

It looks like we need these flags in order to have the symbols and DWARF info for using delve for core analysis:
https://go.googlesource.com/vscode-go/+/HEAD/docs/debugging.md#starting-a-debug-session-fails-with

trozet commented:
We use them in situations to analyze a core using dlv where a trace is not sufficient. Some info I had added to our upstream doc:

dlv core analysis
Revealed https://github.com/ovn-org/ovn-kubernetes/pull/3315
https://github.com/go-delve/delve
https://www.jetbrains.com/help/go/exploring-go-core-dumps.html#view-go-core-dumps-in-goland
https://gist.github.com/trozet/8fb0149cd727939dcc78c60fdcb38f13

It has been critical to us in uncovering a few complex issues. So while I would agree it is not common, it is instrumental in helping us figure out things when all else fails. If I weigh the trade off of ~20MB of the ovnkube binary to the value of being able to figure out a critical bug for a customer then I have to go with the latter.

We could perhaps allow the flags for binaries other than ovnkube since I don't think we have ever used dlv to fix a bug in those.

zshi-redhat commented:
> We use them in situations to analyze a core using dlv where a trace is not sufficient. Some info I had added to our upstream doc:
> 
> dlv core analysis Revealed #3315 https://github.com/go-delve/delve https://www.jetbrains.com/help/go/exploring-go-core-dumps.html#view-go-core-dumps-in-goland https://gist.github.com/trozet/8fb0149cd727939dcc78c60fdcb38f13
> 
> It has been critical to us in uncovering a few complex issues. So while I would agree it is not common, it is instrumental in helping us figure out things when all else fails. If I weigh the trade off of ~20MB of the ovnkube binary to the value of being able to figure out a critical bug for a customer then I have to go with the latter.
> 
agree
> We could perhaps allow the flags for binaries other than ovnkube since I don't think we have ever used dlv to fix a bug in those.
makes sense to me.
updated the PR to apply LDFLAGS to ovnk binaries other than ovnkube.
